qiskit_experiments/data_processing/nodes.py

```diff
@@ -56,130 +106,266 @@ def _format_data(self, datum: Any) -> Any:
                 or averaged IQ date (two-dimensional).
 
         Returns:
-            datum as a numpy array.
+            datum and any error estimate as a numpy array.
 
         Raises:
             DataProcessorError: If the datum does not have the correct format.
         """
         datum = np.asarray(datum, dtype=float)
 
-        if self._validate and len(datum.shape) != self._required_dimension():
-            raise DataProcessorError(
-                f"Single-shot data given {self.__class__.__name__}"
-                f"must be a {self._required_dimension()}D array. Instead, a {len(datum.shape)}D "
-                f"array was given."
-            )
+        if error is not None:
+            error = np.asarray(error, dtype=float)
 
-        return datum
+        if self._validate:
+            if len(datum.shape) not in self._required_dimension():
+                raise DataProcessorError(
+                    f"IQ data given to {self.__class__.__name__} must be an N dimensional"
+                    f"array with N in {self._required_dimension()}. Instead, a {len(datum.shape)}D "
+                    f"array was given."
+                )
+
+            if error is not None and len(error.shape) not in self._required_dimension():
+                raise DataProcessorError(
+                    f"IQ data error given to {self.__class__.__name__} must be an N dimensional"
+                    f"array with N in {self._required_dimension()}. Instead, a {len(error.shape)}D"
+                    f" array was given."
+                )
+
+            if error is not None and len(error.shape) != len(datum.shape):
+                raise DataProcessorError(
+                    "Datum and error do not have the same shape: "
+                    f"{len(datum.shape)} != {len(error.shape)}."
+                )
+
+        return datum, error
 
     def __repr__(self):
         """String representation of the node."""
         return f"{self.__class__.__name__}(validate: {self._validate}, scale: {self.scale})"
 
 
-class ToReal(IQPart):
-    """IQ data post-processing. Isolate the real part of single-shot IQ data."""
+class SVD(IQPart):
+    """Singular Value Decomposition of averaged IQ data."""
 
-    def _required_dimension(self) -> int:
-        """Require memory to be a 3D array."""
-        return 3
+    def __init__(self, validate: bool = True):
+        """
+        Args:
+            validate: If set to False the DataAction will not validate its input.
+        """
+        super().__init__(validate=validate)
+        self._main_axes = None
+        self._means = None
+        self._scales = None
 
-    def _process(self, datum: np.array) -> np.array:
-        """Take the real part of the IQ data.
+    def _required_dimension(self) -> Set[int]:
```

Comment:
Seems like this can be hard-coded as a class variable (if no other child classes override this method to do data processing).

Reply:
Yes see here: 24e0d3b